earth_enterprise/src/fusion/autoingest/plugins.src/RasterProject.src

```diff
@@ -440,7 +440,7 @@ void RasterProjectAssetImplD::HandleEditRequest(
 
   // Custom FindMake - modified to set/preserve indexVersion & fuid_channel_.
   MutableRasterProjectAssetD project =
-      RasterProjectFactory::Find(projname, req.type);
+      Find(projname, req.type, kProjectSubtype);
```

Comment:
One way to do this is to make Find a template function, like this:
```
template <class Type>
Type Find(const std::string &ref_ )
{
    try {
        Asset asset(ref_);
        if (asset &&
            (asset->type == AssetDefs::Map) &&
            (asset->subtype == Type::Name())) {
            return Type(ref_);
        }
    } catch (...) {
        // do nothing - don't even generate any warnings
    }
    return Type();
}
```
Then you'd have to add a new autogenerated function `Name` that returns the type name. You would call it like this:
```
MutableRasterProjectAssetD project =
      Find<MutableRasterProjectAssetD>(projname, req.type);
```
There's still the question of whether this is the right way to approach enhancing the storage manager, but I'm curious to see if this would work.

Reply:
It does make the code simpler and smaller and eliminate some autogeneration, though.